File: sources/global/util/src/main/java/com/tars/util/net/ws/WebsocketHub.java
package com.tars.util.net.ws;

public final class WebsocketHub {

  public static WebsocketServer server(int port, String path) {
    return new Server(port, path);
  }

  public static WebsocketClient client(String address) {
    return new IndependentClient(address);
  }
}

File: sources/global/util/src/main/kotlin/util/network/websocket/WebsocketHub.kt
package util.network.websocket

import com.tars.util.net.ws.WebsocketClient
import com.tars.util.net.ws.WebsocketHub
import com.tars.util.net.ws.WebsocketServer
import util.maid

class WebsocketHub {
    fun server(port: Int, path: String): WebsocketServer {
        return WebsocketHub.server(port, path)
                .apply { maid.internalAdd({ this.stop() }, 1) }
    }

    fun client(address: String): WebsocketClient {
        return WebsocketHub.client(address)
                .apply { maid.internalAdd({ this.stop() }, 1) }
    }
}